examples/handle-head-request/src/main.rs

```diff
@@ -0,0 +1,82 @@
+//! Run with
+//!
+//! ```not_rust
+//! cd examples && cargo run -p example-handle-head-reqeust
+//! ```
+
+use axum::response::{IntoResponse, Response};
+use axum::{http, routing::get, Router};
+use std::net::SocketAddr;
+
+fn app() -> Router {
+    Router::new().route("/get-head", get(get_head_handler))
+}
+
+#[tokio::main]
+async fn main() {
+    let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
+
+    axum::Server::bind(&addr)
+        .serve(app().into_make_service())
+        .await
+        .unwrap();
+}
+
+// get routes will also be called for HEAD requests but will have the response body removed
+// can handle implicit head method request by extract `http::Method` from request
+async fn get_head_handler(method: http::Method) -> Response {
+    // it usually makes sense to handle only special-case HEAD
```

Comment:
Wording can still be improved a bit, I think I suggested something similar before?

```suggestion
    // it usually only makes sense to special-case HEAD
```

Reply:
Yes, thank you.
I just did not notice that.